cli/command/environment_test.go

```diff
@@ -0,0 +1,145 @@
+package command
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/quintilesims/layer0/cli/printer"
+	"github.com/quintilesims/layer0/cli/resolver/mock_resolver"
+	"github.com/quintilesims/layer0/client/mock_client"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/urfave/cli"
+)
+
+func TestCreateEnvironment(t *testing.T) {
+	client, _, envCommand, ctrl := initEnvCommandTest(t)
+	defer ctrl.Finish()
+
+	userData := "user_data"
+	file, close := tempFile(t, userData)
+	defer close()
+
+	createEnvRequest := models.CreateEnvironmentRequest{
+		EnvironmentName:  "name",
+		InstanceSize:     "m3.large",
+		MinClusterCount:  2,
+		UserDataTemplate: []byte(userData),
+		OperatingSystem:  "linux",
+		AMIID:            "ami",
+	}
+
+	environment := &models.Environment{}
+	job := NewJob()
+
+	client.EXPECT().
+		CreateEnvironment(createEnvRequest).
+		Return(job.JobID, nil)
+
+	client.EXPECT().
+		ReadJob(job.JobID).
+		Return(job, nil)
+
+	client.EXPECT().
+		ReadEnvironment(job.Result).
+		Return(environment, nil)
+
+	flags := map[string]interface{}{
+		"size":      createEnvRequest.InstanceSize,
+		"min-count": createEnvRequest.MinClusterCount,
+		"user-data": file.Name(),
+		"os":        createEnvRequest.OperatingSystem,
+		"ami":       createEnvRequest.AMIID,
+	}
+
+	c := GetCLIContext(t, []string{"name"}, flags)
+	if err := envCommand.create(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestCreateEnvironment_userInputErrors(t *testing.T) {
+	_, _, envCommand, ctrl := initEnvCommandTest(t)
+	defer ctrl.Finish()
+
+	contexts := map[string]*cli.Context{
+		"Missing NAME arg": GetCLIContext(t, nil, nil),
+	}
+
+	for name, c := range contexts {
+		if err := envCommand.create(c); err == nil {
+			t.Fatalf("%s: error was nil!", name)
+		}
+	}
+}
+
+func TestDeleteEnvironment(t *testing.T) {
+	client, resolver, envCommand, ctrl := initEnvCommandTest(t)
+	defer ctrl.Finish()
+
+	job := NewJob()
+
+	resolver.EXPECT().
+		Resolve("environment", "name").
+		Return([]string{"id"}, nil)
+
+	client.EXPECT().
+		DeleteEnvironment("id").
+		Return(job.JobID, nil)
+
+	client.EXPECT().
+		ReadJob(job.JobID).
+		Return(job, nil)
+
+	c := GetCLIContext(t, []string{"name"}, nil)
+	if err := envCommand.delete(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestDeleteEnvironmentNoWait(t *testing.T) {
+	client, resolver, envCommand, ctrl := initEnvCommandTest(t)
+	defer ctrl.Finish()
+
+	job := NewJob()
+
+	resolver.EXPECT().
+		Resolve("environment", "name").
+		Return([]string{"id"}, nil)
+
+	client.EXPECT().
+		DeleteEnvironment("id").
+		Return(job.JobID, nil)
+
+	c := GetCLIContext(t, []string{"name"}, map[string]interface{}{"no-wait": true})
```

Comment:
[nit] at one point we had type aliases for args and flags, and it made this function easier to read. I think we should add it back, e.g.:
```
// in something like test_helpers.go
type args []string
type flags map[string]interface{}

// then it reads better in tests
c := getCLIContext(t, args{"name"}, flags{"no-wait": true})
```